### PR TITLE
Remove the :remove named block and yield a :chip block instead

### PR DESCRIPTION
**Pull request by ynotdraw**
## 🚀 Description

This PR removes the `:remove` named block in favor of a `:chip` named block.  This new block gives us more flexibility for new use cases and allows for the following:

- Exposes the `Chip` component directly to consumers
- Attributes to be spread onto the underlying `Chip` component
- Allows consumers to wrap the chip contents in whatever component they need.  For example, if they want to truncate the text of a long select option, they can do `<TruncateText>{{chip.option}}</TruncateText>` and we don't need to support it internally

---

## 🔬 How to Test

- Green build

---

## 📸 Images/Videos of Functionality

- No visual changes, only component API changes

### CODE REVIEW

File: docs/components/multiselect/demo/base-demo.md [hunk 1]
```diff
@@ -149,8 +158,8 @@ export default class extends Component {
     return value === ''
       ? this.options
       : this.options.filter((option) =>
-        option.toLowerCase().includes(value.toLowerCase())
-      );
+          option.toLowerCase().includes(value.toLowerCase())
+        );
```
Comment (ynotdraw) [inline docs/components/multiselect/demo/base-demo.md:162]: Prettier seemed to want this 🤷

File: docs/toucan-form/changeset-validation/demo/base-demo.md [hunk 2]
```diff
@@ -43,46 +43,13 @@
...
-
-      <:remove as |remove|>
-        <remove.Remove @label={{(concat 'Remove' ' ' remove.option.label)}} />
-      </:remove>
-
-      <:default as |multiselect|>
-        <multiselect.Option>{{multiselect.option.label}}</multiselect.Option>
-      </:default>
-    </form.Multiselect>
-
```
Comment (ynotdraw) [inline docs/toucan-form/changeset-validation/demo/base-demo.md:76]: I did a bad job at rebasing yesterday, so this fixes some of those issues.

File: docs/toucan-form/changeset-validation/demo/base-demo.md [hunk 3]
```diff
@@ -181,7 +144,7 @@ export default class extends Component {
       label: 'Teal',
       value: 'teal',
     },
-  ];
+  ].map(({ label }) => label);
```
Comment (clintcs) [inline docs/toucan-form/changeset-validation/demo/base-demo.md:147]: TY

**Review by clintcs — APPROVED** [review]

**Comment by clintcs** [comment]
Real nice!

File: docs/toucan-form/changeset-validation/demo/base-demo.md [hunk 4]
```diff
@@ -43,46 +43,13 @@
...
-    </form.Multiselect>
-
     <form.FileInput
       @label='Files to attach'
       @trigger='Select files'
       @deleteLabel='Delete'
       @name='files'
     />
 
-    <form.Checkbox @label='Agree to the Terms' @name='terms' />
```
[the remark this answers is not in the document]
Comment (ynotdraw) [reply docs/toucan-form/changeset-validation/demo/base-demo.md:84]: haha yeah, unfortunately I botched the rebase and didn't re-review properly so on `main` there's actually two of these.  So these changes remove the dupes 🙃 

- https://github.com/CrowdStrike/ember-toucan-core/blob/main/docs/toucan-form/changeset-validation/demo/base-demo.md?plain=1#L84
- https://github.com/CrowdStrike/ember-toucan-core/blob/main/docs/toucan-form/changeset-validation/demo/base-demo.md?plain=1#L123
- https://github.com/CrowdStrike/ember-toucan-core/blob/main/docs/toucan-form/changeset-validation/demo/base-demo.md?plain=1#L59-L75
- https://github.com/CrowdStrike/ember-toucan-core/blob/main/docs/toucan-form/changeset-validation/demo/base-demo.md?plain=1#L99-L114

**Pull request merged**